homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,181 @@
+            "name": self.name,
+            "date": self.date,
+            "size": self.size,
+            "path": self.path.as_posix(),
+        }
+
+
+class BackupManager:
+    """Backup manager for the Backup integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the backup manager."""
+        self._backups: dict[str, Backup] = {}
+        self.hass = hass
+        self.backup_dir = Path(hass.config.path("backups"))
+        self.backing_up = False
+
+    @property
+    def backups(self) -> dict[str, Backup]:
+        """Return a dict of backups."""
+        return self._backups
+
+    async def load_backups(self) -> None:
+        """Load data of stored backup files."""
+        backups = {}
+
+        def _read_backup_info(backup_path: Path) -> None:
+            with tarfile.open(backup_path, "r:") as backup_file:
+                if data_file := backup_file.extractfile("./backup.json"):
+                    data = json.loads(data_file.read())
+                    backup = Backup(
+                        slug=data["slug"],
+                        name=data["name"],
+                        date=data["date"],
+                        path=backup_path,
+                        size=round(backup_path.stat().st_size / 1_048_576, 2),
+                    )
+                    backups[backup.slug] = backup
+
+        await asyncio.gather(
+            *[
+                self.hass.async_add_executor_job(_read_backup_info, backup)
+                for backup in self.backup_dir.glob("*.tar")
+            ]
+        )
+
+        LOGGER.info("Loaded %s backups", len(backups.values()))
+        self._backups = backups
+
+    def get_backup(self, slug: str) -> Backup | None:
+        """Return a backup."""
+        return self._backups.get(slug)
+
+    def remove_backup(self, slug: str) -> None:
+        """Remove a backup."""
+        if (backup := self.get_backup(slug)) is None:
+            return
+        backup.path.unlink(missing_ok=True)
+        LOGGER.info("Removed backup located at %s", backup.path)
+        self._backups.pop(slug, None)
+
+    async def generate_backup(self) -> str:
+        """Generate a backup."""
+        if self.backing_up:
+            raise HomeAssistantError("Backup in progress")
+
+        self.backing_up = True
+        backup_name = f"Core {HA_VERSION_OBJ}"
+        date_str = now().isoformat()
+        slug = _generate_slug(date_str, backup_name)
+
+        def _create_backup() -> None:
+            with TemporaryDirectory() as tmp_dir:
+                tar_file = Path(self.backup_dir, f"{slug}.tar")
+
+                json_util.save_json(
+                    Path(tmp_dir, "backup.json").as_posix(),
+                    _generate_backup_data(slug, backup_name, date_str, HA_VERSION_OBJ),
+                )
+
+                with tarfile.open(Path(tmp_dir, "homeassistant.tar.gz"), "w:gz") as tar:
+                    _add_directory_to_tarfile(
```

Comment:
We have a feature recently added by @agners to pause the recorder while performing a backup. We should enable that here too.

Reply:
That is planed, but is currently only triggerable with WS, so need a little refactor of the recorder integration before that can be done, which is why that is not including this PR

Reply:
ok. We should generalize that logic indeed, to require an override from the user before we shut down if a backup is in progress